### ISSUE TITLE
Ability to save "Own Profession" for future re-use

### ISSUE THREAD

**Issue by ZeSandman** [elyukai/optolith-client#216]
**Is your feature request related to a problem? Please describe.**
> Not a problem per-se, but related to the issue of profession in newer books that isn't yet in Optolith

**Describe the solution you'd like**
> A way to save a custom profession created using the "own" choice to be saved for re-use (ie: saved as a template).  It would facilitate the use of Newer-but-not-included Profession, as well as Custom ones.

**Describe alternatives you've considered**
> The only alternative I can think of is to manually re-enter everything by hand

**Additional context**
>

**Comment by elyukai** [comment]
So what you want is, you create a hero, add stuff and then click "Save as profession" or sth like that so it can be used as a profession? Do you have a different view on how this feature should be and feel like?

**Comment by ZeSandman** [comment]
That is about it, except only saving what makes the profession what it is, no bits of the Races or Culture. 
Maybe a "Create & Save" profession button?  The useful bit comes from the fact that in beaing able to re-use the "profession" parts.

**Comment by ZeSandman** [comment]
Just thought about something... if we could somehow change the race/culture, we could save the "character" as a template, but that could mean a lot of "template characters" would be hogging the character screen.  It might be easier to do (not a coder so can't really tell) until the Profession are incorporated in Optolith

Or a way to "Import Profession", (with a way to delete them because we're human and forget/make errors)

**Comment by elyukai** [comment]
Yeah, I thought of only the profession bits, too, I'm sorry if I did not make it clear! 😅🙈🙈

Changing race/culture is even something a few already asked for, maybe I can work on that more sooner than later &ndash; but still, if you look at the roadmap, there's still plenty of stuff to do, and I can't work on this project full-time, so I need to prioritize things. 😕 And those are the ones currently sceduled for specific versions in the roadmap. But I'm going to create actual issues for them so I won't forget them! 😅🙈

I'm not quite sure what you mean by "Import Profession"! 🙈

**Comment by elyukai** [comment]
Oh, the issue already exists! 😅 #53

**Comment by ZeSandman** [comment]
Well, I was thinking of a way to add all that is included in a profession... but since I do not really know how you manage your data I put in "somehow import profession info" :)

it might be with a form entry like when you create a new Item, it might be via CSV or something like that, I have no specific idea.

I had seen #53, but it leaves the problem of having too many "Heroes" saved as placeholder templates

**Comment by elyukai** [comment]
Ah, you mean to import a profession from outside the app! Well, that would be an option, too, but I think, saving a character (without race and culture) as a profession package would be way easier to use interface-wise.

Yeah, that's true, I just added it so that anyone who wants to contribute specifically to the stopgap can do that.

**Comment by ZeSandman** [comment]
You're the coder of this tool, we throw suggestions, you choose how to implement them they way it's easier for you.

The less hurdles users have to use it, the more it will be used.  Even if it's a stopgap measure

**Comment by elyukai** [comment]
I mean, I try to exclude implementation details at first and only think of request in terms of user experience: Which option would be the easiest and most intuitive?

True!

As I can't really promise when I will have time to implement this, I can say, that I'm gonna think of this feature in the meantime and maybe another feature requires work on similar parts, in that case I will already prepare some things for this. Thanks for all your input! ❤
